src/webgpu/api/validation/resource_usages/texture/in_render_common.spec.ts

```diff
@@ -230,3 +230,124 @@ g.test('subresources_from_same_texture_as_color_attachment_and_in_bind_group')
       encoder.finish();
     }, !success);
   });
+
+g.test('subresources_from_same_texture_as_depth_stencil_attachment_and_in_bind_group')
+  .desc(
+    `
+  Test that when one subresource of a texture is used as a depth stencil attachment, it cannot be
+  used in a bind group simultaneously in the same render pass encoder. It is allowed when the bind
+  group is used in another render pass encoder instead of the same one.`
+  )
+  .params(u =>
+    u
+      .combine('depthStencilAttachmentLevel', [0, 1])
+      .combine('depthStencilAttachmentLayer', [0, 1])
+      .combineWithParams([
+        { bindGroupViewBaseLevel: 0, bindGroupViewLevelCount: 1 },
+        { bindGroupViewBaseLevel: 1, bindGroupViewLevelCount: 1 },
+        { bindGroupViewBaseLevel: 1, bindGroupViewLevelCount: 2 },
+      ])
+      .combineWithParams([
+        { bindGroupViewBaseLayer: 0, bindGroupViewLayerCount: 1 },
+        { bindGroupViewBaseLayer: 1, bindGroupViewLayerCount: 1 },
+        { bindGroupViewBaseLayer: 1, bindGroupViewLayerCount: 2 },
+      ])
+      .combine('depthStencilReadOnly', [true, false])
+      .combine('bindGroupAspect', ['depth-only', 'stencil-only'] as const)
+      .combine('inSamePass', [true, false])
+  )
+  .fn(async t => {
+    const {
+      depthStencilAttachmentLevel,
+      depthStencilAttachmentLayer,
+      bindGroupViewBaseLevel,
+      bindGroupViewLevelCount,
+      bindGroupViewBaseLayer,
+      bindGroupViewLayerCount,
+      depthStencilReadOnly,
+      bindGroupAspect,
+      inSamePass,
+    } = t.params;
+
+    const viewDimension = bindGroupViewLayerCount > 1 ? '2d-array' : '2d';
+    const bindGroupLayoutEntry: GPUBindGroupLayoutEntry = {
+      binding: 0,
+      visibility: GPUShaderStage.FRAGMENT,
+      texture: { viewDimension, sampleType: bindGroupAspect === 'depth-only' ? 'depth' : 'uint' },
+    };
+    const texture = t.device.createTexture({
+      format: 'depth24plus-stencil8',
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.TEXTURE_BINDING,
+      size: [kTextureSize, kTextureSize, kTextureLayers],
+      mipLevelCount: kTextureLevels,
+    });
+    const bindGroupView = texture.createView({
+      dimension: bindGroupViewLayerCount > 1 ? '2d-array' : '2d',
```

Comment:
nit: we could also always use a 2d-array, just that sometimes it contains a single layer.